src/or/proto_socks.c

```diff
@@ -287,6 +403,45 @@ fetch_from_buf_socks(buf_t *buf, socks_request_t *req,
     buf_clear(buf);
     res = 1;
     goto end;
+  } else if (socks_version == 5) {
+    if (datalen < 2) { /* version and another byte */
+      res = 0;
+      goto end;
+    }
+
+    if (!req->got_auth) {
+
```

Comment:
I'm not sure what's going on here; perhaps it will become clear later in the branch.